src/config/config.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+type Configuration struct {
+	//cast onto the needed type when processing in algos
+	FileInputOption           string `json:"dataOption"`
+	StaticFile                string `json:"staticFilePath"`
+	OutlierDetectionAlgorithm string `json:"outlierDetectionAlgorithm"`
+	MadParameters             string `json:"madParameters"`
+	Prefix                    string `json:"prefix"` // can input a list of string with values seperated by a comma
+	Asn                       string `json:"asn"`
+	PeerIP                    string `json:"peerIP"`
+	Connector                 string `json:"connector"`
+}
+
+func LoadConfig(filename string) (*Configuration, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	var config Configuration
+
+	if err := json.Unmarshal(data, &config); err != nil {
+		return nil, err
+	}
+	return &config, nil
+}
+
+//to parse input of prefix, asn, peer ip, and connector if needed
+func parseByComma(data string) []string {
+	return strings.Split(data, ",")
+}
+
+func ValidDateConfiguration(config *Configuration) {
```

Comment:
Does this function need to be exported? Maybe it makes sense to call this from within LoadConfig function? Im assuming someone will only load a configuration once at the start of the program. But if that is not the case - disregard this.